dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionControllerTest.java

```diff
@@ -17,415 +17,488 @@
+
 import org.apache.dolphinscheduler.api.enums.Status;
-import org.apache.dolphinscheduler.api.service.ProcessDefinitionVersionService;
-import org.apache.dolphinscheduler.api.service.impl.ProcessDefinitionServiceImpl;
-import org.apache.dolphinscheduler.api.utils.PageInfo;
 import org.apache.dolphinscheduler.api.utils.Result;
-import org.apache.dolphinscheduler.common.Constants;
 import org.apache.dolphinscheduler.common.enums.ReleaseState;
-import org.apache.dolphinscheduler.common.enums.UserType;
+import org.apache.dolphinscheduler.common.utils.JSONUtils;
 import org.apache.dolphinscheduler.dao.entity.ProcessDefinition;
 import org.apache.dolphinscheduler.dao.entity.ProcessDefinitionVersion;
-import org.apache.dolphinscheduler.dao.entity.Resource;
-import org.apache.dolphinscheduler.dao.entity.User;
+import org.apache.dolphinscheduler.dao.mapper.ProcessDefinitionMapper;
+import org.apache.dolphinscheduler.dao.mapper.ProcessDefinitionVersionMapper;
 
-import java.text.MessageFormat;
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-
-import javax.servlet.http.HttpServletResponse;
 
 import org.junit.Assert;
-import org.junit.Before;
+import org.junit.FixMethodOrder;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.junit.runners.MethodSorters;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
+import com.fasterxml.jackson.core.type.TypeReference;
+
 
 /**
  * process definition controller test
  */
-@RunWith(MockitoJUnitRunner.Silent.class)
-public class ProcessDefinitionControllerTest {
-
-    private static Logger logger = LoggerFactory.getLogger(ProcessDefinitionControllerTest.class);
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ProcessDefinitionControllerTest extends AbstractControllerTest {
+    private static final Logger logger = LoggerFactory.getLogger(ProcessDefinitionControllerTest.class);
 
-    @InjectMocks
-    private ProcessDefinitionController processDefinitionController;
+    public static String projectName;
+    public static String projectId;
+    public static String definitionId;
 
-    @Mock
-    private ProcessDefinitionServiceImpl processDefinitionService;
+    @Autowired
+    private ProcessDefinitionVersionMapper processDefinitionVersionMapper;
```

Comment:
If you remove @Mock, this will affect the unit test when there is no database, I am not sure. It does not seem to be using DB2 at the moment

Reply:
many controller tests use a database,  it does not matter.